### ISSUE TITLE
Password JSON output is cut off when using CLI with Nodejs

### ISSUE THREAD

**Issue by tmwrnr** [Dashlane/dashlane-cli#235]
**Describe the bug**
Since the new feature "user presence verification with biometrics" (#231) was introduced, you can not fetch the complete output from the CLI via Nodejs, if a lot of passwords are stored.

For getting all passwords (`dcli p --output json`) with `childProcess.execFile()` you would have to use the `maxBuffer` option. Otherwise the printed output of the JSON is cut off after **8190 characters**.

With the commit https://github.com/Dashlane/dashlane-cli/commit/e16b13793957410a11471d614a5c0c0ee1fd385a everything works, but with the next commit https://github.com/Dashlane/dashlane-cli/commit/0507f57fdad949c9596179aec62accd7541bdc5c it seems that `maxBuffer` is ignored.

**To Reproduce**

1. Create a file with the following content:
```js
const childProcess = require("node:child_process");
const util = require("node:util");

const execFile = util.promisify(childProcess.execFile);

const CLI_PATH = "path-to-cli";
const MAX_BUFFER = 4096 * 1024;

execFile(CLI_PATH, ["p", "--output", "json"], { maxBuffer: MAX_BUFFER }).then(({ stdout, stderr }) => {
  if (stderr) throw new Error(stderr);

  console.log("Length: ", stdout.length);
  console.log("Last characters", stdout.slice(-10));
});
```
2. Checkout https://github.com/Dashlane/dashlane-cli/commit/e16b13793957410a11471d614a5c0c0ee1fd385a, compile CLI and run the Node file
3. The stdout should be a **valid** JSON
4. Checkout https://github.com/Dashlane/dashlane-cli/commit/0507f57fdad949c9596179aec62accd7541bdc5c, compile CLI and run the Node file
5. The stdout should be a **invalid** JSON and have a different `length`

**Expected behavior**

I would expect, that maxBuffer has an effect and can be used to get to full stdout.

**Environment (please complete the following information):**
 - OS: macOS 14.4.1
 - Version: [v6.2415.0](https://github.com/Dashlane/dashlane-cli/releases/tag/v6.2415.0)
 - Node Version: v20.10.0

**Additional context**

The new feature was discussed in https://github.com/raycast/extensions/pull/11326.

**Comment by Mikescops** [comment]
Hello,

Thanks for reporting this issue, I'll spend time on it next week when I'm back.

**Comment by Mikescops** [comment]
Hello,

So I have spent quite some times deep diving this issue.

I know what line to blame between the two commits.

```ts
program
    .parseAsync()
    .catch((error: Error) => {
        console.error(errorColor(`error: ${error.message}`));
        process.exit(1);
    })
    .finally(() => process.exit(0)); // <-- here
```

The reason I added this is to make sure the program exit once the command is run, that way any potential thread hanging in the background is closed.
When I made my tests it was working just fine from the terminal but indeed as you reported the `execFile` is not able to buffer the output.

Took me a while to figure out, but using `process.exit()` terminates the child program instantly, and it doesn't wait for any I/O to complete, including the stdout. It seems then that the exit is faster that the stdout write+flush

There are a couple of issues mentioning this:
- https://github.com/nodejs/node/issues/784
- https://github.com/nodejs/node/issues/19218

What I've come around is to write an empty buffer to the stdout and use the callback function to exit once it's done, this will force to wait for the previous write+flush to end and then it can exit.

```js
process.stdout.write('', 'utf-8', () => process.exit(0))
```

I know it feels like a monkey patch but it's the best I could come up with for now.
